mlflow/models/evaluation/base.py

```diff
@@ -957,6 +951,26 @@ def _validate(validation_thresholds, candidate_metrics, baseline_metrics=None):
     raise ModelValidationFailedException(message=os.linesep.join(failure_messages))
 
 
+def _convert_data_to_mlflow_dataset(data, targets=None, dataset_path=None):
+    """Convert input data to mlflow dataset."""
+    if "pyspark" in sys.modules:
+        from pyspark.sql import DataFrame as SparkDataFrame
+
+    if isinstance(data, Dataset):
+        return data
+    elif isinstance(data, pd.DataFrame):
+        return mlflow.data.from_pandas(df=data, targets=targets, source=dataset_path)
+    elif "pyspark" in sys.modules and isinstance(data, SparkDataFrame):
+        return mlflow.data.from_spark(df=data, targets=targets, path=dataset_path)
+    elif isinstance(data, np.ndarray):
+        return mlflow.data.from_numpy(data, targets=targets, source=dataset_path)
```

Comment:
what is the type of the dataset source? 

as discussed offline, a `CodeDatasetSource` will give a pointer to the code, which imo is the best effort source without further context. however, i'm open to other options if we think it makes more sense.

Reply:
done!